### ISSUE TITLE
Auto generate list of external dependencies

### ISSUE THREAD

**Issue by marcoscaceres** [w3c/respec#2560]
It would be nice ReSpec outputted a a list of terms cited by reference... and, if we know which what the type of a term is, we labelled the term correctly. For example:

"The HTML spec defines the following concepts: queue a task. It also defines the following elements: `<body>`. And it defines the following IDL fragments: `HTMLWhateverThing` interface, and the `Foo` dictionary."

**Comment by sidvishnoi** [comment]
`xref` adds a `data-type` attribute on all the links, so we just need to collect all terms by `data-cite` and then by `data-type` and render them. 

What about the links which editors added without using xref? Should we show them as "misc" ("xyz also defines a, b, c") or ignore them?

From the current structure of code, this all has to be done before calling `linkInlineCitations` (in `core/link-to-dfn`), as it includes a cleanup step which removes `data-cite*` stuff (which I think should be done in end of all plugins).

Perhaps we should do a refactoring of link-to-dfn.js and data-cite.js first.

**Comment by marcoscaceres** [comment]
> What about the links which editors added without using xref? Should we show them as "misc" ("xyz also defines a, b, c") or ignore them?

We should add them... but we need to look at conventions of how specs do this manually already. For example:
https://fetch.spec.whatwg.org/#index

Basically, the sections:
"Terms defined by this specification"
"Terms defined by reference"

**Comment by ridhishjain** [comment]
@marcoscaceres @sidvishnoi I would like to work on this. Seems challenging, a little help regarding understanding this more thoroughly would be great!

**Comment by sidvishnoi** [comment]
As Marcos said "we need to look at conventions of how specs do this manually already". So, presently, we need to do some research how other documents try to list terms defined externally, and then decide which one we want to support. 
Some examples of what we're trying to generate:
https://w3c.github.io/manifest/#terms-defined-by-reference
https://w3c.github.io/sensors/#index-defined-elsewhere
https://w3c.github.io/payment-method-basic-card/#dependencies
https://fetch.spec.whatwg.org/#index-defined-elsewhere

So, we need to find more examples and figure out what we want to support. Only after that, we'll look into implementation.

**Comment by sidvishnoi** [comment]
Taking following features from examples above:

- https://w3c.github.io/manifest/#terms-defined-by-reference
  - links to the [reference] also 
- https://fetch.spec.whatwg.org/#index-defined-elsewhere, https://w3c.github.io/sensors/#index-defined-elsewhere
  - back references to usage in current document
  - looks clean
- https://w3c.github.io/sensors/#index-defined-elsewhere
  - column layout saves some space 

Some terms also tell about the "for context", like "for each (for map)". It's a nice to have - if we can make it less noisy (not showing where it doesn't make sense).

Should we define types along with terms? Like "AbortSignal interface", "allow attribute". Need to make it as less noisy as possible.

**Comment by sidvishnoi** [comment]
Another idea is to roll our own, as described in the issue description, i.e. no lists/tables but natural language.

**Comment by marcoscaceres** [comment]
>  Like "AbortSignal interface", "allow attribute". 

Yes, that would be awesome. If we know the type that would be amazing!

**Comment by marcoscaceres** [comment]
> i.e. no lists/tables but natural language.

Imagine this serves as a kind of "book index"... so we could also look at how books do this. So we do want a list, I think. And it's nice to scan terms. Some alternatives might be to order them in different ways: like alphabetically, and then put the spec after each. That would make it more useful.

**Comment by marcoscaceres** [comment]
Looking at fetch, what BikeShed does with classification is actually pretty good (specially in the local ones)... and adding the type info (and appropriate style) would make it even better + plus making it multi-column.
